From 734764272e328114801815a06b2c6abe0a4dea4c Mon Sep 17 00:00:00 2001
From: Sergio Martins <smartins@kde.org>
Date: Thu, 11 Jun 2020 11:44:55 +0100
Subject: [PATCH] Make clazy build 30% faster by using precompiled headers

Requires cmake >= 3.16

Doing it for the base class of each check (checkbase.h) seems
already good enough.
---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 514ba47..5317244 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -142,6 +142,11 @@ macro(add_clang_plugin name)
 
   add_library(${name} SHARED ${srcs})
 
+  if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.16.0")
+      # 30% speedup
+      target_precompile_headers(${name} PRIVATE src/checkbase.h)
+  endif()
+
   if(SYMBOL_FILE)
     set_target_properties(${name} PROPERTIES LINK_FlAGS "-exported_symbols_list ${SYMBOL_FILE}")
   endif()
-- 
2.27.0

